### PR TITLE
Add `pytest` and `jest` configuration

### PR DESCRIPTION
**Pull request by trungleduc**
This PR adds configuration files and example tests for `jest` and `pytest` to the template.
A new test workflow is also added.

### CODE REVIEW

**Comment by martinRenou** [comment]
Thanks!

File: {{cookiecutter.github_project_name}}/.github/workflows/tests.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  run:
+    shell: bash -l {0}
+
+jobs:
+  tests:
+    name: Test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: [3.10]
```
Comment (jtpio) [inline {{cookiecutter.github_project_name}}/.github/workflows/tests.yml]: Looks like we might need to do something similar to https://github.com/jupyterlab-contrib/jupyterlab-app-cookiecutter/pull/21 here too?

```suggestion
        python: ["3.10"]
```

Comment (trungleduc) [reply {{cookiecutter.github_project_name}}/.github/workflows/tests.yml]: Thanks!

**Comment by jtpio** [comment]
It would also be useful to create a new app on CI with this cookiecutter and the `cookiecutter` CLI, and try to run the tests, just to make sure they run fine.

It could be in https://github.com/jupyterlab-contrib/jupyterlab-app-cookiecutter/blob/main/.github/workflows/tests.yml, or in a new workflow.

Similar to what the JupyterLab extension cookiecutter does https://github.com/jupyterlab/extension-cookiecutter-ts/blob/3.0/.github/workflows/main.yml

**Comment by trungleduc** [comment]
> It would also be useful to create a new app on CI with this cookiecutter and the `cookiecutter` CLI, and try to run the tests, just to make sure they run fine.
> 
> It could be in https://github.com/jupyterlab-contrib/jupyterlab-app-cookiecutter/blob/main/.github/workflows/tests.yml, or in a new workflow.
> 
> Similar to what the JupyterLab extension cookiecutter does https://github.com/jupyterlab/extension-cookiecutter-ts/blob/3.0/.github/workflows/main.yml

Thanks @jtpio  for the suggestion! Added in  the last commit.

**Comment by jtpio** [comment]
Nice, thanks!

Looks like we should either add `.pytest_cache` to `prettierignore`, or move the testing step after the lint:

image

**Review by jtpio — APPROVED** [review]
CI is now passing. 

Many thanks @trungleduc for this!

**Comment by jtpio** [comment]
A follow-up could be to split the tests into separate jobs, so it's not all in "Visual Regression".

**Pull request merged**